prow/cmd/deck/main.go

```diff
@@ -167,7 +168,7 @@ func gatherOptions(fs *flag.FlagSet, args ...string) options {
 	fs.BoolVar(&o.dryRun, "dry-run", false, "Whether or not to make mutating API calls to GitHub.")
 	fs.StringVar(&o.pluginConfig, "plugin-config", "", "Path to plugin config file, probably /etc/plugins/plugins.yaml")
 	o.kubernetes.AddFlags(fs)
-	o.github.AddFlagsWithoutDefaultGitHubTokenPath(fs)
+	o.github.AddFlags(fs)
```

Comment:
Don't we _want_ no default? We should do the opposite ofthis

Reply:
Not sure I fully understand what you are getting at. I changes this, because we can not use an anonymous client anymore, because during construction of the `GitClient` from the `GitHub` options an API call is made that fails if there is no token: https://github.com/kubernetes/test-infra/blob/f94d49fff34c3e8b7ec234f89792de8515291dee/prow/flagutil/github.go#L136

Hypothetically we could change that thought, the `GitHub` client already has code so it can be used without a token. WDYT?

Reply:
Steve is correct. We want people to explicitly set a --github-token-path flag, not depend on there magically happening to be a file at `/etc/github/oauth`

https://github.com/kubernetes/test-infra/blob/9fab55afee2561cdcdaeaa67c59c30d9926a00db/prow/flagutil/github.go#L55-L59

Reply:
Reverted that part, ptal